src/contracts/middleware/StakeRegistry.sol

```diff
@@ -0,0 +1,481 @@
+                ++quorumNumbersIndex;
+            }
+        }
+    }
+
+    /**
+     * @notice Removes the stakes of the operator
+     */
+    function _deregisterOperator(bytes32 operatorId, bytes memory quorumNumbers) internal {
+        uint8 quorumNumbersLength = uint8(quorumNumbers.length);
+        // check the operator is deregistering from only valid quorums
+        require(uint8(quorumNumbers[quorumNumbersLength - 1]) < quorumCount, "StakeRegistry._registerStake: greatest quorumNumber must be less than quorumCount");
+        OperatorStakeUpdate memory _operatorStakeUpdate;
+        // add the `updateBlockNumber` info
+        _operatorStakeUpdate.updateBlockNumber = uint32(block.number);
+        OperatorStakeUpdate memory _newTotalStakeUpdate;
+        // add the `updateBlockNumber` info
+        _newTotalStakeUpdate.updateBlockNumber = uint32(block.number);
+        // loop through the operator's quorums and remove the operator's stake for each quorum
+        for (uint8 quorumNumbersIndex = 0; quorumNumbersIndex < quorumNumbersLength;) {
+            uint8 quorumNumber = uint8(quorumNumbers[quorumNumbersIndex]);
+            // update the operator's stake
+            uint96 stakeBeforeUpdate = _recordOperatorStakeUpdate(operatorId, quorumNumber, _operatorStakeUpdate);
+            // subtract the amounts staked by the operator that is getting deregistered from the total stake before deregistration
+            // copy latest totalStakes to memory
+            _newTotalStakeUpdate.stake = _totalStakeHistory[quorumNumber][_totalStakeHistory[quorumNumber].length - 1].stake - stakeBeforeUpdate;
+            // update storage of total stake
+            _recordTotalStakeUpdate(quorumNumber, _newTotalStakeUpdate);
+
+            emit StakeUpdate(
+                operatorId,
+                quorumNumber,
+                // new stakes are zero
+                0
+            );
+            unchecked {
+                ++quorumNumbersIndex;
+            }
+        }
+    }
+
+    /**
+     * @notice Finds the updated stake for `operator`, stores it and records the update.
+     * @dev **DOES NOT UPDATE `totalStake` IN ANY WAY** -- `totalStake` updates must be done elsewhere.
+     */
+    function _updateOperatorStake(address operator, bytes32 operatorId, uint8 quorumNumber)
+        internal
+        returns (uint96, uint96)
+    {   
+        // determine new stakes
+        OperatorStakeUpdate memory operatorStakeUpdate;
+        operatorStakeUpdate.updateBlockNumber = uint32(block.number);
+        operatorStakeUpdate.stake = weightOfOperator(quorumNumber, operator);
+
+        // check if minimum requirements have been met
+        if (operatorStakeUpdate.stake < minimumStakeForQuorum[quorumNumber]) {
+            operatorStakeUpdate.stake = uint96(0);
+        }
+        // initialize stakeBeforeUpdate to 0
+        uint96 stakeBeforeUpdate = _recordOperatorStakeUpdate(operatorId, quorumNumber, operatorStakeUpdate);
```

Comment:
the comment here seems incorrect?

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/2631769ce9169012dc2edc79ac9c7834381de6d6